api/agent/stats.go

```diff
@@ -310,11 +314,36 @@ func RegisterContainerViews(tagKeys []string, latencyDist []float64) {
 			continue
 		}
 		v := common.CreateView(containerGaugeMeasures[i], view.Sum(), tagKeys)
+
+		if err := view.Register(v); err != nil {
+			logrus.WithError(err).Fatal("cannot register view")
+		}
+	}
+
+	for i, key := range containerGaugeKeys {
+		if key == "" {
+			continue
+		}
+		v := common.CreateViewWithTags(containerGaugeMeasures[i], view.Sum(), containerKeys)
+
 		if err := view.Register(v); err != nil {
 			logrus.WithError(err).Fatal("cannot register view")
 		}
 	}
 
+
+	for i, key := range containerGaugeKeys {
+		if key == "" {
+			continue
+		}
+		v := common.CreateView(containerGaugeMeasures[i], view.Sum(), tagKeys)
+
+		if err := view.Register(v); err != nil {
+			logrus.WithError(err).Fatal("cannot register view")
+		}
+	}
```

Comment:
this block is a duplicate of https://github.com/fnproject/fn/pull/1373/files#diff-4eaa01cd93e12bd2b4e3fb22e40fc312R316 - we can remove it. I can't recall the semantics here exactly it looks like we're registering 2 views of the container gauges with different tags, does opencensus merge these for us? can we simply join the tags together and create the view one time for behavior clarity? I'm also curious if this works when we make metrics that don't have all the tags (as we would if we don't enable the new tags), will the view have the tags but empty or are the tags not there (this would possibly create upstream prometheus issues)? if the tags only show up in prometheus when the metrics fill them in, then I think we can add these tags to all the container metrics in here, as well (since this will only happen when configured)?

Reply:
The short answer is that a registered view is only applicable for the metric with the tags registered for the view. I did some testing with this and if the view with tags is registered (and the view without tags isn't) then the metrics will only show up if all the tags are turned on. It's pretty annoying but this seemed like the best solution.